code/game/machinery/doors/firedoor.dm

```diff
@@ -317,7 +318,13 @@
 
 /obj/machinery/door/firedoor/close()
 	latetoggle()
-	return ..()
+	..()
+
+	if(push_on_close)
+		for(var/mob/living/L in loc)
+			for(var/dir in global.cardinal)
+				if(L.Move(get_step(loc, dir)))
+					break
```

Comment:
This isn't going to stop firedoors from closing on top of people if there's nowhere to push them.

Reply:
Could make it fail to close if it can't push, if we want to double down on this change.